test/python/compiler/test_transpiler.py

```diff
@@ -757,6 +757,53 @@ def test_translation_method_synthesis(self, optimization_level, basis_gates):
         self.assertTrue(Operator(out).equiv(qc))
         self.assertTrue(set(out.count_ops()).issubset(basis_gates))
 
+    def test_transpiled_custom_gates_calibration(self):
+        """Test if transpiled calibrations is equal to custom gates circuit calibrations."""
+        custom_180 = Gate("mycustom", 1, [3.14])
+        custom_90 = Gate("mycustom", 1, [1.57])
+
+        circ = QuantumCircuit(2)
+        circ.append(custom_180, [0])
+        circ.append(custom_90, [1])
+
+        with pulse.build() as q0_x180:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), pulse.DriveChannel(0))
+        with pulse.build() as q1_y90:
+            pulse.play(pulse.library.Gaussian(20, -1.0, 3.0), pulse.DriveChannel(1))
+
+        # Add calibration
+        circ.add_calibration(custom_180, [0], q0_x180)
+        circ.add_calibration(custom_90, [1], q1_y90)
+
+        backend = FakeAlmaden()
+        transpiled_circuit = transpile(
+            circ,
+            backend=backend,
+            basis_gates=backend.configuration().basis_gates
+            + list(circ.calibrations.keys()),
+        )
```

Comment:
Add a TODO that we have to remove lines 782-783 (in the next PR)

Reply:
Added in https://github.com/Qiskit/qiskit-terra/pull/5049/commits/98ae90ffb5bf534ca41c72b1ce5ab465c7f687e3